server/builtin.go

```diff
@@ -35,6 +42,10 @@ var secretsManagerBackends = map[secrets.SecretsManagerType]secrets.SecretsManag
 	secrets.GCPSSM:         gcpssm.SecretsManagerFactory,
 }
 
+var genesisCreationFactory = map[ConsensusType]GenesisFactoryHook{
```

Comment:
Why is this defined as a global variable?

Reply:
We have just followed the pattern with
 `var consensusBackends = map[ConsensusType]consensus.Factory{`
 and
 `var secretsManagerBackends = map[secrets.SecretsManagerType]secrets.SecretsManagerFactory{`
 
 but we can write logic
 ```
 // custom write genesis hook per consensus engine
	engineName := m.config.Chain.Params.GetEngine()
	if factory, exists := genesisCreationFactory[ConsensusType(engineName)]; exists {
		m.executor.GenesisPostHook = factory(m.config.Chain, engineName)
	}
```
inside some function and remove variable. What do you think?